twake/backend/node/src/services/files/web/controllers/files.ts

```diff
@@ -0,0 +1,40 @@
+import { FastifyRequest, FastifyReply } from "fastify";
+import { CrudController } from "../../../../core/platform/services/webserver/types";
+import {
+  ResourceCreateResponse,
+  ResourceDeleteResponse,
+  ResourceGetResponse,
+  ResourceListResponse,
+} from "../../../../services/types";
+import { File } from "../../entities/file";
+import { FileServiceAPI } from "../../api";
+import fs from "fs";
+import util from "util";
+import { pipeline } from "stream";
+
+const pump = util.promisify(pipeline);
+
+export class FileController {
+  constructor(protected service: FileServiceAPI) {}
+
+  async save(request: FastifyRequest<{}>, response: FastifyReply) {
+    const data = await request.file();
+    data.file;
+    data.fields;
+    data.fieldname;
+    data.filename;
+    data.encoding;
+    data.mimetype;
+
+    await pump(data.file, fs.createWriteStream(data.filename));
```

Comment:
If I am not wrong, you are streaming the file to nowhere.
From the documentation (https://nodejs.org/api/stream.html#stream_stream_pipeline_source_transforms_destination_callback):

```
The pipeline API provides a promise version, which can also receive an options argument as the last parameter with a signal <AbortSignal> property. When the signal is aborted, destroy will be called on the underlying pipeline, with an AbortError.
```

Using the `AbortController ` will also be nice to handle aborted uploads